detect.go

```diff
@@ -0,0 +1,81 @@
+package composer
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/paketo-buildpacks/packit/v2"
+	"github.com/paketo-buildpacks/packit/v2/fs"
+	"github.com/paketo-buildpacks/packit/v2/scribe"
+)
+
+// PhpVersionResolverInterface provides a Resolve method to determine which Php Version to request
+//go:generate faux --interface PhpVersionResolverInterface --output fakes/php_version_resolver.go
+type PhpVersionResolverInterface interface {
+	Resolve(composerJsonPath, composerLockPath string) (version, versionSource string, err error)
+}
+
+func Detect(logEmitter scribe.Emitter, phpVersionResolver PhpVersionResolverInterface) packit.DetectFunc {
+	return func(context packit.DetectContext) (packit.DetectResult, error) {
+		composerJsonPath, composerLockPath, composerVar, composerVarFound := FindComposerFiles(context.WorkingDir)
```

Comment:
```suggestion
		composerJsonPath, composerLockPath, _, composerVarFound := FindComposerFiles(context.WorkingDir)
```

for clarity, since `composerVar` isn't used anywhere